README.md

```diff
@@ -1,25 +1,67 @@
 
-Transforming CSV records into JSON files is an optional feature intended for cases where a consumer of the DEX CSV product would rather work with individual JSON records than CSV files.
-
-Each transformed record is assigned a UUIDv4 and is hashed. The intent behind hashing each record is to aid downstream CDC data systems in detecting duplicate records. The file hash and UUIDv4, among other properties, are included in the file metadata when it's written to disk.
-
-The CSV product, when built, will be fully integrated with the rest of DEX and will be optional for any teams wishing to process CSV files through DEX. More information will be forthcoming as the CSV products moves from proof-of-concept into the Alpha and Beta phases of product maturity.
-
-## Character encoding
-The CSV code modules will optionally accept one of several character encodings, such as `UTF-8` or `ISO-8859-1`, and can auto-detect character encodings (albeit with no guarantee of 100% accuracy) if one is not provided by the calling function. 
+## Key Features
+
+- **CSV/TSV validation** - The validator, which checks the files for their adherence to the rules set forth in [RFC 4180](https://www.rfc-editor.org/rfc/rfc4180). The CSV validator does not validate field content, header fields or data types at this point. Each validated row is assigned:
+    - **File UUID**: Each processed file is assigned a unique `uuid` that is used to link rows to its originating file. This will enable downstream consumers to track back rows to its source file.
+    - **Row UUID**: Each row is assigned unique `uuid` during validation. This ensure that every row can be uniquely referenced.
+    - **SHA-256 Row Hashing**: The content of each row is hashed using  `SHA-256` algorithm to provide ability of detecting duplicates and ensure data integrity throughout the data pipeline. 
+
+- **CSV/TSV transformer** - The transformer, which generates a JSON object for each row in the file. If the optional header is present, then the header specification is used to populate the JSON property names.
+
+- **Encoding Detection** - If encoding is not specified in the optional command line argument `config.json` , the tool will sample the file for up to 1024 bytes, and attempt to `auto-detect` the file's encoding (with best effort accuracy). Suppported encodings are: 
+    - `UTF-8`
+    - `UTF-8 with BOM`
+    - `USASCII`
+    - `ISO-8859-1`
+    - `Windows1252`
+
+- **Delimiter Detection** - If the delimiter is not provided in optional `config.json` command file flag, the tool will sample the file upto 1024 bytes of data, and will attempt to `auto-detect` delimiter. Supported delimiters include: 
+    - `,` (comma)
+    - `\t` (tab) 
+
+## Installation
+1. Clone the repository:
+    ```bash
+    git clone https://github.com/CDCgov/data-exchange-csv.git
+    cd data-exchange-csv
+    go build
+    ```
+
+## Usage
+The DEX CSV Validator and Transformer accepts following command-line flags:
```

Comment:
Can you provide an example command syntax? Don't want to guess what the default executable or command is (e.g. `./data-exchange-csv -fileURL <path> -destination <path>`).